### ISSUE TITLE
In PHP 7 and 7.1, JSON decode errors in the SDK's PHP library on DELETE requests

### ISSUE THREAD

**Issue by benlk** [npr/pmp-php-sdk#37]
Example occurrences, from npr/pmp-wordpress-plugin's test suite:

- WordPress 4.8 and PHP 7: https://travis-ci.org/npr/pmp-wordpress-plugin/jobs/370241638
- WordPress 4.9 and PHP 7: https://travis-ci.org/npr/pmp-wordpress-plugin/jobs/370241639
- WordPress 4.8 and PHP 7.1: https://travis-ci.org/npr/pmp-wordpress-plugin/jobs/370241640
- WordPress 4.9 and PHP 7.1: https://travis-ci.org/npr/pmp-wordpress-plugin/jobs/370241641

This issue overlaps with and may be caused by https://github.com/npr/pmp-wordpress-plugin/issues/133, so this might not be the proper place to report it.

```
PHP Fatal error:  Uncaught Pmp\Sdk\Exception\RemoteException: [204]: JSON decode error => DELETE https://publish-sandbox.pmp.io/docs/9d11c322-6f0b-41c8-9ffa-5a401d999b2a
  thrown in phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/Http.php on line 179
Fatal error: Uncaught Pmp\Sdk\Exception\RemoteException: [204]: JSON decode error => DELETE https://publish-sandbox.pmp.io/docs/9d11c322-6f0b-41c8-9ffa-5a401d999b2a
  thrown in phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/Http.php on line 179
Pmp\Sdk\Exception\RemoteException: JSON decode error in phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/Http.php on line 179
Call Stack:
    0.0058     507384   1. {main}() /home/travis/.phpenv/versions/7.0.25/bin/phpunit:0
    0.1799    9252832   2. PHPUnit\TextUI\Command::main() /home/travis/.phpenv/versions/7.0.25/bin/phpunit:563
    0.1799    9261624   3. PHPUnit\TextUI\Command->run() phar:///home/travis/.phpenv/versions/7.0.25/bin/phpunit/phpunit/TextUI/Command.php:148
    1.7386   36455592   4. PHPUnit\TextUI\TestRunner->doRun() phar:///home/travis/.phpenv/versions/7.0.25/bin/phpunit/phpunit/TextUI/Command.php:195
    1.7457   36460456   5. PHPUnit\Framework\TestSuite->run() phar:///home/travis/.phpenv/versions/7.0.25/bin/phpunit/phpunit/TextUI/TestRunner.php:537
   42.1291   45311032   6. PHPUnit\Framework\TestSuite->run() phar:///home/travis/.phpenv/versions/7.0.25/bin/phpunit/phpunit/Framework/TestSuite.php:743
   50.0098   45894232   7. PMP_SyncerTestCase::tearDownAfterClass() phar:///home/travis/.phpenv/versions/7.0.25/bin/phpunit/phpunit/Framework/TestSuite.php:748
   50.3253   46086840   8. Pmp\Sdk\CollectionDocJson->delete() /tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/tests/PMPSyncerTestCase.php:44
   50.3256   46087520   9. Pmp\Sdk\CollectionDocJson->_request() phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/CollectionDocJson.php:219
```

**Comment by benlk** [comment]
No, this is not caused by https://github.com/npr/pmp-wordpress-plugin/issues/133.

In test run https://travis-ci.org/npr/pmp-wordpress-plugin/jobs/370638705 with PHP 7.1 and WordPress 4.9, the constant `PMP_NOTIFICATIONS_SECRET` was defined as a fixed string instead of using the output of the `crypt()` function, which was causing errors to be thrown. Errors with `crypt()` are the cause of https://github.com/npr/pmp-wordpress-plugin/issues/133. If by removing `crypt()` from the execution flow, this problem was not solved, we can conclude that `crypt()` was not the problem.

```
PHP Fatal error:  Uncaught Pmp\Sdk\Exception\RemoteException: [204]: JSON decode error => DELETE https://publish-sandbox.pmp.io/docs/3c97d758-6aaa-4597-b4ef-cfefcc2df351
  thrown in phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/Http.php on line 179
Fatal error: Uncaught Pmp\Sdk\Exception\RemoteException: [204]: JSON decode error => DELETE https://publish-sandbox.pmp.io/docs/3c97d758-6aaa-4597-b4ef-cfefcc2df351
  thrown in phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/Http.php on line 179
Pmp\Sdk\Exception\RemoteException: JSON decode error in phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/Http.php on line 179
Call Stack:
    0.0022     507384   1. {main}() /home/travis/.phpenv/versions/7.1.11/bin/phpunit:0
    0.0862    9232312   2. PHPUnit\TextUI\Command::main() /home/travis/.phpenv/versions/7.1.11/bin/phpunit:563
    0.0863    9241104   3. PHPUnit\TextUI\Command->run() phar:///home/travis/.phpenv/versions/7.1.11/bin/phpunit/phpunit/TextUI/Command.php:148
    2.2605   37508952   4. PHPUnit\TextUI\TestRunner->doRun() phar:///home/travis/.phpenv/versions/7.1.11/bin/phpunit/phpunit/TextUI/Command.php:195
    2.2692   37579096   5. PHPUnit\Framework\TestSuite->run() phar:///home/travis/.phpenv/versions/7.1.11/bin/phpunit/phpunit/TextUI/TestRunner.php:537
   27.0783   43218968   6. PHPUnit\Framework\TestSuite->run() phar:///home/travis/.phpenv/versions/7.1.11/bin/phpunit/phpunit/Framework/TestSuite.php:743
   39.8099   46352448   7. PMP_SyncerTestCase::tearDownAfterClass() phar:///home/travis/.phpenv/versions/7.1.11/bin/phpunit/phpunit/Framework/TestSuite.php:748
   40.0854   46391536   8. Pmp\Sdk\CollectionDocJson->delete() /tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/tests/PMPSyncerTestCase.php:44
   40.0856   46392184   9. Pmp\Sdk\CollectionDocJson->_request() phar:///tmp/wordpress/src/wp-content/plugins/pmp-wordpress-plugin/vendor/pmpsdk.phar/Pmp/Sdk/CollectionDocJson.php:219
```

**Comment by scottsmith130** [comment]
Should be resolved by this commit: https://github.com/npr/pmp-php-sdk/commit/de8859f1621386a6a2107c7cdda142c4ea96fbe5

**Comment by scottsmith130** [comment]
This should be resolved in latest v2.0.0 release
